zio-sbt-githubactions/src/main/scala/zio/sbt/githubactionsnative/ScalaWorkflow.scala

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2022-2023 dev.zio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.sbt.githubactionsnative
+
+import zio.json._
+import zio.sbt.githubactionsnative.ScalaWorkflow.JavaVersion.JDK11
+
+// The original code of the githubactions package was originally copied from the zio-aws-codegen project:
+// https://github.com/zio/zio-aws/tree/master/zio-aws-codegen/src/main/scala/zio/aws/codegen/githubactions
+object ScalaWorkflow {
```

Comment:
What's need to rewrite this code from scratch?

Reply:
I added the new solution next to the existing. But I think in the end result the old model and workflow are not used anymore in the other modules. 
Perhaps the question should be: should we use the new model, stick with the old or somehow support both? 
My reasons for the new model is that it is closer to writing direct GHA yaml and less of a cognitive load (no need to learn another DSL). It also simplifies the codecs because we reduce the amount of codec adjustments or required DTO's.

Reply:
I would love to contribute some more but I really think a model which is as much 1:1 with GHA model as possible, helps to build better workflows. It is predictable and the codecs are easier.